tests/lambda_functions/va_profile/test_va_profile_remove_old_opt_outs.py

```diff
@@ -0,0 +1,102 @@
+from sqlalchemy import text
+from datetime import datetime, timedelta
+
+INSERT_OPT_IN_OUT_RECORD = text(
+    """INSERT INTO va_profile_local_cache(va_profile_id, communication_item_id,
+    communication_channel_id, source_datetime, allowed)
+    VALUES(:va_profile_id, :communication_item_id, :communication_channel_id,
+    :source_datetime, :allowed)"""
+)
+
+REMOVE_OPTED_OUT_RECORDS_QUERY = text("""SELECT va_profile_remove_old_opt_outs();""")
+
+COUNT = r"""SELECT COUNT(*) FROM va_profile_local_cache;"""
+
+COUNT_OPTED_OUT_RECORD_QUERY = text("""SELECT COUNT(*)
+FROM va_profile_local_cache
+WHERE allowed = False
+AND age(NOW(), source_datetime) > INTERVAL '24 hours';""")
+
+
+def setup_db(connection):
+    """
+    Using the given connection, truncate the VA Profile local cache, and call
+    the stored procedure to add a specific row. This establishes a known state
+    for testing. Truncating is necessary because the database side effects of
+    executing the VA Profile lambda function are not rolled back at the
+    conclusion of a test.
+    """
+
+    connection.execute("truncate va_profile_local_cache;")
+
+    # Sanity check
+    count_queryset = connection.execute(COUNT)
+    assert count_queryset.fetchone()[0] == 0, "The cache should be empty at the start."
+
+    expired_datetime = (datetime.now() - timedelta(days=2)).strftime('%Y-%m-%dT%H:%M:%S%z')
+
+    insert_expired_record_opt_out = INSERT_OPT_IN_OUT_RECORD.bindparams(
+        va_profile_id=0,
+        communication_item_id=0,
+        communication_channel_id=0,
+        allowed=False,
+        source_datetime=expired_datetime
+    )
+
+    connection.execute(insert_expired_record_opt_out)
+
+    insert_expired_record_opt_in = INSERT_OPT_IN_OUT_RECORD.bindparams(
+        va_profile_id=2,
+        communication_item_id=3,
+        communication_channel_id=4,
+        allowed=True,
+        source_datetime=expired_datetime
+    )
+
+    connection.execute(insert_expired_record_opt_in)
+
+    insert_active_record_opt_out = INSERT_OPT_IN_OUT_RECORD.bindparams(
+        va_profile_id=1,
+        communication_item_id=2,
+        communication_channel_id=3,
+        allowed=False,
+        source_datetime=datetime.now().strftime('%Y-%m-%dT%H:%M:%S%z')
+    )
+
+    connection.execute(insert_active_record_opt_out)
+
+    count_records_currently_in_database = connection.execute(COUNT)
+    assert count_records_currently_in_database.fetchone()[0] == 3, \
+        "There should only be three records in the database."
+
+
+def test_count_opted_out_records_query(notify_db_session):
```

Comment:
This is a sanity check of your test code (setup_db) rather than a test of the lambda implementation.  I would move the assertion to the next test and delete this test.